web/src/server_events_dispatch.js

```diff
@@ -283,7 +283,21 @@ export function dispatch_normal_event(event) {
                     switch (event.property) {
                         case "default":
                             for (const [key, value] of Object.entries(event.data)) {
-                                page_params["realm_" + key] = value;
+                                if (key === "authentication_methods") {
+                                    for (const [auth_method, enabled] of Object.entries(
+                                        event.data.authentication_methods,
+                                    )) {
+                                        page_params.realm_authentication_methods[
+                                            auth_method
+                                        ].enabled = enabled;
+                                    }
+                                    settings_org.populate_auth_methods(
+                                        event.data.authentication_methods,
+                                    );
```

Comment:
This is needed because `page_params.realm_authentication_methods` looks like `{"Email": {"enabled": True, "available": True}, ...}`, but the event has the simple form `{"Email": True, ...}`, so we need to correctly apply these bool values from the event  to `.enabled`

Reply:
Hmm, shouldn't we just update the event format to have the exact same structure as this comes in `register`? That would allow us to live-update these other properties too.